src/query-builder/InsertQueryBuilder.ts

```diff
@@ -347,6 +347,18 @@ export class InsertQueryBuilder<Entity> extends QueryBuilder<Entity> {
             query += ` RETURNING ${returningExpression}`;
         }
 
+
+        // Inserting a specific value for an auto-increment primary key in mssql requires enabling IDENTITY_INSERT
+        // IDENTITY_INSERT can only be enabled for tables where there is an IDENTITY column and only if there is a value to be inserted (i.e. supplying DEFAULT is prohibited if IDENTITY_INSERT is enabled)
+        if (this.connection.driver instanceof SqlServerDriver
+            && this.expressionMap.mainAlias!.hasMetadata
+            && this.expressionMap.mainAlias!.metadata.columns
+                .filter((column) => this.expressionMap.insertColumns.length > 0 ? this.expressionMap.insertColumns.indexOf(column.propertyPath) !== -1 : column.isInsert)
+                .some((column) => this.isOverridingAutoIncrementBehavior(column))
+        ) {
+            query = `SET IDENTITY_INSERT ${tableName} ON; ${query}; SET IDENTITY_INSERT ${tableName} OFF`;
```

Comment:
Are you able to escape the path for the table name here?

Reply:
Have a look at line 304. Either the table name is already escaped, or there is an issue with injection already.